src/librustc_yk_sections/mir_cfg.rs

```diff
@@ -98,10 +96,16 @@ impl<'a, 'tcx, 'gcx> ConvCx<'a, 'tcx, 'gcx> {
     /// Add `bb` as a definition site of the TIR variable `var`.
     fn push_def_site(&self, bb: BasicBlock, var: TirVarIndex) {
         let mut sites = self.def_sites.borrow_mut();
-        if sites.len() <= usize::try_from(var).unwrap() {
-            sites.resize((var + 1) as usize, Vec::new());
+        // Vector indices are `usize`, but variable indices are `u32`, so converting from a
```

Comment:
I would probably delete this comment, and then put (above the `let var_usize` line) a short comment along the lines of "var is u32 which the earlier assert guarantees can fit into a usize".

Reply:
Delete the assertion too, I think?

Reply:
Only if the assertion was 100% guaranteed to have been made my some other bit of code (and, if so, we should document it).

Reply:
The assertion is identical to the earlier one you referenced.

But let me suggest we leave it in, since it protects from the other being removed later, and we've established that these assertions are pretty much free. What do you think?

Reply:
Yes, I know it's identical, but *is it guaranteed to have been called before this code is executed*?

Reply:
Ah sorry. I see.

Yes, the variable will have to have been created prior to it's def site being pushed.